documents/payment.md

```diff
@@ -630,6 +630,196 @@ Doc reference [doc](https://razorpay.com/docs/payments/payment-methods/cards/aut
+### Create Payment UPI s2s / VPA token (Third party validation)
+
+```rb
+param_attr = {
+  "amount": 200,
+  "currency": "INR",
+  "order_id": "order_GAWRjlWkVcRh0V",
+  "email": "gaurav.kumar@example.com",
+  "contact": "9123456789",
+  "method": "upi",
+  "customer_id": "cust_EIW4T2etiweBmG",
+  "save": 1,
+  "ip": "192.168.0.103",
+  "referer": "http",
+  "user_agent": "Mozilla/5.0",
+  "description": "Test flow",
+  "notes": {
+    "note_key": "value1"
+  },
+  "upi": {
+    "flow": "collect",
+    "vpa": "gauravkumar@exampleupi",
+    "expiry_time": 5
+  }
+}
+
+Razorpay::Payment.create_upi(para_attr)
+```
+
+**Parameters:**
+| Name        | Type    | Description                          |
+|-------------|---------|--------------------------------------|
+| amount*          | integer | Amount of the order to be paid  |
+| currency*   | string  | The currency of the payment (defaults to INR)                                  |
+| order_id*        | string  | The unique identifier of the order created. |
+| email*        | string      | Email of the customer                       |
+| customer_id*   | string      | The id of the customer to be fetched |
+| contact*      | string      | Contact number of the customer              |
+| notes | array  | A key-value pair  |
+| description | string  | Descriptive text of the payment. |
+| save | boolean  |  Specifies if the VPA should be stored as tokens.Possible value is `0`, `1`  |
+| callback_url   | string      | URL where Razorpay will submit the final payment status. |
+| ip*   | string      | The client's browser IP address. For example `117.217.74.98` |
+| referer*   | string      | Value of `referer` header passed by the client's browser. For example, `https://example.com/` |
+| user_agent*   | string      | Value of `user_agent` header passed by the client's browser. For example, `Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36` |
+| upi* (for Upi only)  | array      | All keys listed [here](https://razorpay.com/docs/payments/third-party-validation/s2s-integration/upi/collect#step-14-initiate-a-payment) are supported  |
+
+**Response:** <br>
+```json
+{
+  "razorpay_payment_id": "pay_EAm09NKReXi2e0"
+}
+```
+-------------------------------------------------------------------------------------------------------
+### Create Payment UPI s2s / VPA token (Third party validation)
+
+```rb
+param_attr = {
+    "amount": 100,
+    "currency": "INR",
+    "order_id": "order_Ee0biRtLOqzRjP",
+    "email": "gaurav.kumar@example.com",
+    "contact": "9090909090",
+    "method": "upi",
+    "ip": "192.168.0.103",
+    "referer": "http",
+    "user_agent": "Mozilla/5.0",
+    "description": "Test flow",
+    "notes": {
+      "purpose": "UPI test payment"
+    },
+    "upi": {
+      "flow" : "intent"
+    }
+}
+
+Razorpay::Payment.create_upi(para_attr)
```

Comment:
Error on create_upi
Also change variable name from para_attr to param_attr

Reply:
try with these order id's . i will share api cred for this

`order_JryIs7T7t7zeWV, order_JryJAT9qDE6dqZ, order_JryJHXlh4WELs1`

para_attr = {
    "amount": 100,
    "currency": "INR",
    "order_id": "order_JryExkN1YXfugO",
    "email": "gaurav.kumar@example.com",
    "contact": "9090909090",
    "method": "upi",
    "ip": "192.168.0.103",
    "referer": "http",
    "user_agent": "Mozilla/5.0",
    "description": "Test flow",
    "notes": {
      "purpose": "UPI test payment"
    },
    "upi": {
      "flow": "intent"
    }
}

Razorpay::Payment.create_upi(para_attr)